endless/eoswindow.c

```diff
@@ -537,6 +544,46 @@ eos_window_size_allocate (GtkWidget *window, GtkAllocation *allocation)
   GTK_WIDGET_CLASS (eos_window_parent_class)->size_allocate (window, allocation);
 }
 
+static gboolean
+in_resize_timeout (gpointer data)
+{
+  EosWindowPrivate *priv = eos_window_get_instance_private (data);
+
+  gtk_style_context_remove_class (gtk_widget_get_style_context (data),
+                                  "in-resize");
+  priv->in_resize_id = 0;
+
+  return G_SOURCE_REMOVE;
+}
+
+static gboolean
+eos_window_configure_event (GtkWidget *widget, GdkEventConfigure *event)
+{
+  EosWindowPrivate *priv = eos_window_get_instance_private (EOS_WINDOW (widget));
+
+  if (event->width != priv->width || event->height != priv->height)
+    {
+      guint64 current_time = g_source_get_time (g_main_current_source ());
+
+      if (current_time - priv->last_configure < 1000000)
+        {
+          if (priv->in_resize_id)
+            g_source_remove (priv->in_resize_id);
+          else
+            gtk_style_context_add_class (gtk_widget_get_style_context (widget),
+                                         "in-resize");
+
+          priv->in_resize_id = g_timeout_add_seconds (1, in_resize_timeout, widget);
```

Comment:
One second feels a bit too long, and add_seconds() will try and coalesce the timer, which means that it may take a bit more than a second — even an arbitrary amount of time longer than a second, if the kernel is so inclined.

Reply:
I don't have a strong suggestion, here; need to try it out a bit — maybe with a designer.

Reply:
yeah maybe half a second is better, unless you can think of an actual way to know when a window drag/resize starts and ends